http.go

```diff
@@ -98,22 +98,22 @@ func newRouter() *mux.Router {
 	// To maintain compatibility with previous API versions, wrap legacy API
 	// calls to add the type field and route to the correct handler
 	r.HandleFunc("/{value:(?:[0-9]{1,3}\\.){3}[0-9]{1,3}}",
-		auth(wrapLegacyIPRequest(httpGetReputation))).Methods("GET")
+		auth(wrapLegacyIPRequest(httpGetReputation), false)).Methods("GET")
 	r.HandleFunc("/{value:(?:[0-9]{1,3}\\.){3}[0-9]{1,3}}",
-		auth(wrapLegacyIPRequest(httpPutReputation))).Methods("PUT")
+		auth(wrapLegacyIPRequest(httpPutReputation), true)).Methods("PUT")
 	r.HandleFunc("/{value:(?:[0-9]{1,3}\\.){3}[0-9]{1,3}}",
-		auth(wrapLegacyIPRequest(httpDeleteReputation))).Methods("DELETE")
+		auth(wrapLegacyIPRequest(httpDeleteReputation), true)).Methods("DELETE")
 	r.HandleFunc("/violations/{value:(?:[0-9]{1,3}\\.){3}[0-9]{1,3}}",
-		auth(wrapLegacyIPRequest(httpPutViolation))).Methods("PUT")
-	r.HandleFunc("/violations", auth(wrapLegacyIPRequest(httpPutViolations))).Methods("PUT")
-
-	r.HandleFunc("/violations", auth(httpGetViolations)).Methods("GET")
-	r.HandleFunc("/dump", auth(httpGetAllReputation)).Methods("GET")
-	r.HandleFunc("/type/{type:[a-z]{1,12}}/{value}", auth(httpGetReputation)).Methods("GET")
-	r.HandleFunc("/type/{type:[a-z]{1,12}}/{value}", auth(httpPutReputation)).Methods("PUT")
-	r.HandleFunc("/type/{type:[a-z]{1,12}}/{value}", auth(httpDeleteReputation)).Methods("DELETE")
-	r.HandleFunc("/violations/type/{type:[a-z]{1,12}}/{value}", auth(httpPutViolation)).Methods("PUT")
-	r.HandleFunc("/violations/type/{type:[a-z]{1,12}}", auth(httpPutViolations)).Methods("PUT")
+		auth(wrapLegacyIPRequest(httpPutViolation), true)).Methods("PUT")
+	r.HandleFunc("/violations", auth(wrapLegacyIPRequest(httpPutViolations), true)).Methods("PUT")
+
+	r.HandleFunc("/violations", auth(httpGetViolations, false)).Methods("GET")
+	r.HandleFunc("/dump", auth(httpGetAllReputation, false)).Methods("GET")
```

Comment:
Let's change this one to require write access, purely because it can be used to impact the service.

Reply:
Will make the change -- but I disagree that It should require write perms, as it is purely a read operation. I'd argue that you wouldnt give a token to someone who can't see the dump.

Reply:
It's a very expensive operation in redis, and the read only tokens would typically be given to consuming clients. We'd want it to require additional privileges in this case just purely because it can be abused to disable the service, has nothing to do with the contents of the response.

Reply:
Oh! Interesting, had not thought of that. 👍